sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala

```diff
@@ -607,6 +607,38 @@ object DataSourceReadBenchmark extends SqlBasedBenchmark {
     }
   }
 
+  def vectorizedScanPartitionColumnsBenchmark(values: Int, pColumns: Int): Unit = {
```

Comment:
hmm not really sure if this benchmark is necessary. What does it compare?

Reply:
Due to the lack of base, it is not easy to explain the end-to-end performance improvement. 

However, before this pr, Parquet and Orc used the same data structure for the partition column in VectorizedRead. Therefore, from the results of this scenario, Parquet and Orc had the same scan performance before this pr, but Parquet was slightly faster than Orc after this pr. 

ref https://github.com/apache/spark/pull/36616#discussion_r882346084 and https://github.com/apache/spark/pull/36616#discussion_r882346982

I think after a similar work of Orc, they should have the same performance again.

So I added this microbenchmark scenario, if really think it is unnecessary, I can delete it.

Reply:
@sunchao Do you suggest to keep or delete this new benchmark  scene?

Reply:
Thanks. I feel we can just showing the benchmark result in the PR instead of adding it to `DataSourceReadBenchmark`. In this PR we already added dedicated benchmarks for the constant vector so here it seems a bit redundant especially since there is no base case to compare to.

Reply:
OK ~ [c736821](https://github.com/apache/spark/pull/36616/commits/c7368215f593826e670c8d4aab04b316399aba25) revert change of `DataSourceReadBenchmark` and mircrobench result